src/sentry/lang/native/symbolicator.py

```diff
@@ -389,20 +390,16 @@ def parse_backfill_sources(sources_json, original_sources):
     return sources
 
 
-def redact_source_secrets(config_sources):
-    """
-    Returns a json string with all of the secrets redacted from every source.
-
-    May raise InvalidSourcesError if the provided sources are invalid.
-    """
+def redact_source_secrets(config_sources: json.JSONData) -> json.JSONData:
+    """Returns a JSONData with all of the secrets redacted from every source."""
 
-    sources = parse_sources(config_sources, False)
-    for source in sources:
+    redacted_sources = deepcopy(config_sources)
```

Comment:
I *think* a copy is enough (there's `config_sources.copy()` for that these days instead of having to import the module) but this sure makes things easier to reason about and i doubt this is performance-sensitive so this is probably better.